src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+                                Statement.RETURN_GENERATED_KEYS)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+            return book;
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to insert book " + book + " into DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = mapToBook(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to retrieve book with id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY);
+                ResultSet resultSet = statement.executeQuery()) {
+            while (resultSet.next()) {
+                books.add(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB", e);
+        }
+
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BOOK_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            statement.executeUpdate();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            int rowsAffected = statement.executeUpdate();
+            return rowsAffected > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete book with id " + id, e);
+        }
+    }
+
+    Book mapToBook(ResultSet resultSet) throws SQLException {
```

Comment:
When you are converting `ResultSet` to `Book`, it's better to use either setters or a constructor consistently rather than both. In this case, you are using the constructor to set title and price, and then using a setter to set the id. Choose one approach for consistency.